File: samples/src/main/java/net/ickis/deluge/samples/HandlingEvents.java
package net.ickis.deluge.samples;

import io.reactivex.Observable;
import io.reactivex.disposables.Disposable;
import net.ickis.deluge.event.TorrentRemovedEvent;

public class HandlingEvents {
    public static void main(String[] args) {
        SimpleClient.create(client -> {
            Observable<String> observable = client.subscribe(TorrentRemovedEvent.INSTANCE);
            Disposable disposable = observable.subscribe(torrentId -> System.out.println("Removed " + torrentId));
            sleep(10000);
            disposable.dispose();
            return null;
        });
    }

    private static void sleep(long ms) {
        try {
            Thread.sleep(ms);
        } catch (InterruptedException ignore) {}
    }
}

File: samples/src/main/kotlin/net/ickis/deluge/samples/HandlingEvents.kt
package net.ickis.deluge.samples

import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch
import net.ickis.deluge.event.TorrentRemovedEvent

fun main() {
    simpleClient { client ->
        val channel = client.subscribe(TorrentRemovedEvent)
        GlobalScope.launch {
            for (torrentId in channel) {
                println("Removed $torrentId")
            }
        }
        delay(10000)
        channel.cancel() // clean up when channel is no longer needed
    }
}
